--- dedalo.ori	2022-09-23 17:52:56.265619046 +0200
+++ dedalo	2022-09-23 17:53:50.965636196 +0200
@@ -101,16 +101,16 @@
         fi
     ;;
     start)
-        systemctl start dedalo
+        /etc/init.d/dedalo start
     ;;
     restart)
-        systemctl restart dedalo
+        /etc/init.d/dedalo restart
     ;;
     stop)
-        systemctl stop dedalo
+        /etc/init.d/dedalo stop
     ;;
     status)
-        systemctl status dedalo
+        echo "Not implemented!"
     ;;
     info)
         echo "Dedalo: Network Access Controller"
