Index: ns-dedalo-0.0.1/dedalo/template/engine
===================================================================
--- ns-dedalo-0.0.1.orig/dedalo/template/engine
--- ns-dedalo-0.0.1/dedalo/template/engine
@@ -22,7 +22,7 @@
 source /opt/icaro/dedalo/config
 HS_DIGEST=$(echo -n "$HS_SECRET$HS_UUID" | md5sum | awk '{print $1}')
 HS_AAA_HOST=$(echo "$HS_AAA_URL" | sed -e 's|https\?://\([^/:]\+\)\(.*\)|\1|')
-HS_TIMEZONE=$(timedatectl | grep "Time zone" | awk '{print $3}')
+HS_TIMEZONE=$(/sbin/uci get system.@system[0].timezone)
 HS_WALLED_GARDEN_INTEGRATIONS=$(find /opt/icaro/dedalo/walled_gardens/integrations/ -type f -exec echo include {} \;)
 
 eval "cat <<EOF
