diff --git a/net/openvpn/Makefile b/net/openvpn/Makefile
--- a/net/openvpn/Makefile
+++ b/net/openvpn/Makefile
@@ -10,7 +10,7 @@
 PKG_NAME:=openvpn
 
 PKG_VERSION:=2.7.4
-PKG_RELEASE:=2
+PKG_RELEASE:=3
 
 PKG_SOURCE_URL:=\
 	https://build.openvpn.net/downloads/releases/ \
diff --git a/net/openvpn/files/openvpn.init b/net/openvpn/files/openvpn.init
--- a/net/openvpn/files/openvpn.init
+++ b/net/openvpn/files/openvpn.init
@@ -21,6 +21,7 @@ append_param() {
 	local s="$1"
 	local v="$2"
 	case "$v" in
+		*_*_*_*_*) v=${v%%_*}-${v#*_}; v=${v%%_*}-${v#*_}; v=${v%%_*}-${v#*_}; v=${v%%_*}-${v#*_} ;;
 		*_*_*_*) v=${v%%_*}-${v#*_}; v=${v%%_*}-${v#*_}; v=${v%%_*}-${v#*_} ;;
 		*_*_*)   v=${v%%_*}-${v#*_}; v=${v%%_*}-${v#*_} ;;
 		*_*)     v=${v%%_*}-${v#*_} ;;
diff --git a/net/openvpn/files/openvpn.options b/net/openvpn/files/openvpn.options
--- a/net/openvpn/files/openvpn.options
+++ b/net/openvpn/files/openvpn.options
@@ -140,6 +140,7 @@ x509_username_field
 '
 
 OPENVPN_BOOLS='
+allow_deprecated_insecure_static_crypto
 allow_recursive_routing
 auth_nocache
 auth_user_pass_optional
