diff --git i/net/adblock/files/adblock.init w/net/adblock/files/adblock.init
index ed02d69ff..32647663b 100755
--- i/net/adblock/files/adblock.init
+++ w/net/adblock/files/adblock.init
@@ -35,6 +35,7 @@ boot() {

 start_service() {
 	if "${adb_init}" enabled; then
+		/usr/sbin/ts-dns # configure threat shield dns, if needed
 		if [ "${action}" = "boot" ]; then
 			[ -n "$(uci_get adblock global adb_trigger)" ] && return 0
 		fi
@@ -49,15 +50,18 @@ start_service() {
 }

 restart() {
+	/usr/sbin/ts-dns # configure threat shield dns, if needed
 	stop_service "restart"
 	rc_procd start_service restart
 }

 reload_service() {
+	/usr/sbin/ts-dns # configure threat shield dns, if needed
 	rc_procd start_service reload
 }

 stop_service() {
+	/usr/sbin/ts-dns # configure threat shield dns, if needed
 	[ -z "${1}" ] && rc_procd "${adb_script}" stop
 }
 
