diff --git a/net/adblock/files/README.md b/net/adblock/files/README.md
index 95cbfeedc..688bdb7cc 100644
--- a/net/adblock/files/README.md
+++ b/net/adblock/files/README.md
@@ -159,6 +159,7 @@ Available commands:
 | adb_debug          | 0, disabled                        | set to 1 to enable the debug output                                                            |
 | adb_nice           | 0, standard prio.                  | valid nice level range 0-19 of the adblock processes                                           |
 | adb_forcedns       | 0, disabled                        | set to 1 to force DNS requests to the local resolver                                           |
+| adb_bypass         | -, not set                         | list of IP addresses excluded from `adb_forcedns` option                                       |
 | adb_dnsdir         | -, auto-detected                   | path for the generated blocklist file 'adb_list.overall'                                       |
 | adb_dnstimeout     | 10                                 | timeout in seconds to wait for a successful DNS backend restart                                |
 | adb_dnsinstance    | 0, first instance                  | set to the relevant dns backend instance used by adblock (dnsmasq only)                        |
diff --git a/net/adblock/files/adblock.sh b/net/adblock/files/adblock.sh
index e76e81450..c673d7d43 100755
--- a/net/adblock/files/adblock.sh
+++ b/net/adblock/files/adblock.sh
@@ -197,6 +197,9 @@ f_conf() {
 				eval "${option}=\"$(printf "%s" "${adb_zonelist}") ${value}\""
 			elif [ "${option}" = "adb_portlist" ]; then
 				eval "${option}=\"$(printf "%s" "${adb_portlist}") ${value}\""
+			elif [ "${option}" = "adb_bypass" ]
+			then
+				eval "${option}=\"$(printf "%s" "${adb_bypass}") ${value}\""
 			fi
 		}
 	}
@@ -552,9 +555,20 @@ f_extconf() {
 	f_uci "${config}"
 
 	config="firewall"
-	fwcfg="$(uci -qNX show "${config}" | "${adb_awk}" 'BEGIN{FS="[.=]"};/adblock_/{if(zone==$2){next}else{ORS=" ";zone=$2;print zone}}')"
+	fwcfg="$(uci -qNX show "${config}" | grep -v adblock_bypass | "${adb_awk}" 'BEGIN{FS="[.=]"};/adblock_/{if(zone==$2){next}else{ORS=" ";zone=$2;print zone}}')"
 	if [ "${adb_enabled}" = "1" ] && [ "${adb_forcedns}" = "1" ] &&
 		/etc/init.d/firewall enabled; then
+		uci -q batch <<-EOC
+			set firewall."adblock_bypass"="ipset"
+			set firewall."adblock_bypass".name="adblock_bypass"
+			set firewall."adblock_bypass".match="src_net"
+			set firewall."adblock_bypass".enabled="1"
+		EOC
+		uci -q delete firewall."adblock_bypass".entry  # make sure bypass list is always empty
+		for src in ${adb_bypass}
+		do
+			uci -q add_list firewall.adblock_bypass.entry="${src}"
+		done
 		for zone in ${adb_zonelist}; do
 			for port in ${adb_portlist}; do
 				if ! printf "%s" "${fwcfg}" | grep -q "adblock_${zone}${port}[ |\$]"; then
@@ -567,6 +581,7 @@ f_extconf() {
 						set firewall."adblock_${zone}${port}".dest_port="${port}"
 						set firewall."adblock_${zone}${port}".target="DNAT"
 						set firewall."adblock_${zone}${port}".family="any"
+						set firewall."adblock_${zone}${port}".ipset="!adblock_bypass
 					EOC
 				fi
 				fwcfg="${fwcfg/adblock_${zone}${port}[ |\$]/}"
@@ -579,6 +594,7 @@ f_extconf() {
 		for section in ${fwcfg}; do
 			uci_remove firewall "${section}"
 		done
+		uci_remove firewall "adblock_bypass"
 	fi
 	f_uci "${config}"
 }
